ql/src/java/org/apache/hadoop/hive/ql/parse/LoadSemanticAnalyzer.java

```diff
@@ -291,18 +291,43 @@ private void analyzeLoad(ASTNode ast) throws SemanticException {
           ErrorMsg.INVALID_PATH.getMsg(), fromTree, e.getMessage()), e);
     }
 
+    Map<String, String> tmpPartSpec = null;
+    Table table;
+    try {
+      table = db.getTable(HiveTableName.withNoDefault(getUnescapedName((ASTNode) tableTree.getChild(0))));
+    } catch (HiveException e) {
+      throw new SemanticException(
+          ASTErrorUtils.getMsg(ErrorMsg.CANNOT_RETRIEVE_TABLE_METADATA.getMsg(), tableTree.getChild(0), e.getMessage()),
+          e);
+    }
+
+    if (tableTree.getChildCount() == 2 && table.isNonNative() && table.getStorageHandler() != null &&
+        table.getStorageHandler().supportsAppendData(table.getTTable(), true)) {
+      ASTNode partspec = (ASTNode) tableTree.getChild(1);
+      // partSpec is a mapping from partition column name to its value.
+      tmpPartSpec = new HashMap<String, String>(partspec.getChildCount());
+      for (int i = 0; i < partspec.getChildCount(); ++i) {
+        ASTNode partspec_val = (ASTNode) partspec.getChild(i);
+        String val = null;
+        String colName = unescapeIdentifier(partspec_val.getChild(0).getText().toLowerCase());
+        val = stripQuotes(partspec_val.getChild(1).getText());
+        tmpPartSpec.put(colName, val);
+        tableTree.deleteChild(1);
```

Comment:
should this be outside of for loop?

Reply:
same thing is done in `reparseAndSuperAnalyze`

Reply:
I was looking at some different stuff, thanx, have refactored to use the same. I wrote it myself, If I found it before, would have saved efforts :-(